### ISSUE TITLE
PHP Deprecation - fnmatch(): Passing null to parameter #2 ($filename) of type string is deprecated

### ISSUE THREAD

**Issue by csgerg** [mautic/mautic#13270]
### Mautic Version

5.0.x series

### PHP version

8.1.27

### What browsers are you seeing the problem on?

Firefox

### What happened?

After visit the login pages it creates journal entry  in the error_log what is in installation root folder.

Mautic version: 5.0.2

### How can we reproduce this issue?

Open login url: /s/login

### Relevant log output

```shell
PHP Deprecation - fnmatch(): Passing null to parameter #2 ($filename) of type string is deprecated - in file /app/middlewares/CORSMiddleware.php - at line 107
PHP Deprecated:  fnmatch(): Passing null to parameter #2 ($filename) of type string is deprecated in /app/middlewares/CORSMiddleware.php on line 107
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

**Comment by J-Wick4** [comment]
This code got rid of the error in the logs. 
Can someone verify it?

The error message indicates that passing null as the $filename parameter to the fnmatch() function is deprecated. This is happening because the $origin variable may be null in some cases.

To fix this error, you can add a check to ensure that $origin is not null before calling fnmatch(). Here's an updated version of the code:

```
 // Check the domains using shell wildcard patterns
 $validCorsDomainFilter = function ($validCorsDomain) use ($origin) {
     if ($origin === null) {
         return false;
     }
     return fnmatch($validCorsDomain, $origin, FNM_CASEFOLD);
 };

 if (array_filter($this->validCORSDomains, $validCorsDomainFilter)) {
     $this->requestOriginIsValid = true;
     $this->corsHeaders['Vary']  = 'Origin';

     return $origin;
 }

 $this->requestOriginIsValid = false;

 return null;
```
This adds an additional check to ensure that **$origin** is not null before calling **fnmatch()**. If **$origin** is null, the filter will **return false** and the function will continue as before.

**Comment by back-2-95** [comment]
I just setup a new Mautic install for production and our logs are filled with this deprecation notice.

**Comment by RCheesley** [comment]
Looks like we have a fix in #13270 folks - please check and test, and remember to leave your review!

Docs here for how to review & test: https://mau.tc/tester (you can do it in the browser using Gitpod).

Thanks in advance for helping get this into a future release of Mautic!
